### PR TITLE
Fix GCS bucket name validation

### PR DESCRIPTION
**Pull request by jootten**
### Motivation and context

It is not possible to attach a GCS cloud storage bucket which name contains a dot. As stated [here](https://cloud.google.com/storage/docs/buckets#naming), GCS bucket names allow dots. This is a major problem for us, and this PR fixes this issue.

### How has this been tested?

No tests required.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by Marishka17** [comment]
@jootten, Hello, thank you for the contribution!

Looks like it also affects attaching AWS S3 buckets. Could you please add  `.` directly to `COMMON_ALLOWED_RESOURCE_NAME_SYMBOLS`?

**Comment by bsekachev** [comment]
@Marishka17 

Could you please take a look?

File: changelog.d/20231205_204101_jootten_fix_validation.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+### Fixed
+
+- Fix gcs bucket name validation
```
Comment (Marishka17) [inline changelog.d/20231205_204101_jootten_fix_validation.md:3]:
```suggestion
- Attaching GCS and AWS S3 buckets with dots in name
```

File: cvat/apps/engine/serializers.py [hunk 2]
```diff
@@ -1796,15 +1796,15 @@ def validate(self, attrs):
...
-        )
         ALLOWED_RESOURCE_NAME_SYMBOLS = (
-            COMMON_ALLOWED_RESOURCE_NAME_SYMBOLS
-            if provider_type != models.CloudProviderChoice.GOOGLE_CLOUD_STORAGE
-            else COMMON_ALLOWED_RESOURCE_NAME_SYMBOLS + "_"
+            string.ascii_lowercase + string.digits + "-"
         )
 
+        if provider_type == models.CloudProviderChoice.GOOGLE_CLOUD_STORAGE:
+            ALLOWED_RESOURCE_NAME_SYMBOLS += "_" + "."
```
Comment (Marishka17) [inline cvat/apps/engine/serializers.py:1804]:
```suggestion
            ALLOWED_RESOURCE_NAME_SYMBOLS += "_."
```

**Review by Marishka17 — APPROVED** [review]

**Pull request merged**